src/Autofac/Builder/RegistrationData.cs

```diff
@@ -147,17 +147,17 @@ public IComponentLifetime Lifetime
         /// <summary>
```

Comment:
The switch from interface to concrete implementation here is one of the areas I'm a little concerned about. Exposing the set of handlers like this makes any underlying improvement a breaking change.